sklearn/tests/test_nca.py

```diff
@@ -0,0 +1,24 @@
+import numpy as np
+from sklearn.metric_learning import NCA
+from sklearn.utils.testing import ignore_warnings
+from sklearn.utils.testing import assert_array_almost_equal
+
+N = 300
+aux = (np.concatenate([0.5*np.ones((N/2, 1)),
+                       np.zeros((N/2, 1)), 1.1*np.ones((N/2, 1))], axis=1))
+X = np.concatenate([np.random.rand(N/2, 3),
+                    np.random.rand(N/2, 3) + aux])
+
+y = np.concatenate([np.concatenate([np.ones((N/2, 1)), np.zeros((N/2, 1))]),
+                    np.concatenate([np.zeros((N/2, 1)), np.ones((N/2, 1))])],
+                   axis=1)
+X = X.T
+y = y[:, 0]
+A = np.array([[1, 0, 0], [0, 1, 0]])
+
+def test_NCA():
+	# Training
+	nca = NCA.NCA(metric=A, method='BFGS', objective='KL-divergence', options={'maxiter': 10, 'disp': True})
+	print nca.score(X, y)
```

Comment:
Thanks. Yes I will use assertions later. Just wanted to pass Travis first.

This single test is not failing actually. It passed on my local pc. It's
just not being included in __all__, which caused other tests failed in the
package. I'll update soon.
On Sep 16, 2015 4:27 AM, "B@rmaley.exe" <notifications@github.com> wrote:

> In sklearn/tests/test_nca.py
> <https://github.com/scikit-learn/scikit-learn/pull/5276#discussion_r39604544>
> :
>
> > +aux = (np.concatenate([0.5*np.ones((N/2, 1)),
> > +                       np.zeros((N/2, 1)), 1.1*np.ones((N/2, 1))], axis=1))
> > +X = np.concatenate([np.random.rand(N/2, 3),
> > +                    np.random.rand(N/2, 3) + aux])
> > +
> > +y = np.concatenate([np.concatenate([np.ones((N/2, 1)), np.zeros((N/2, 1))]),
> > +                    np.concatenate([np.zeros((N/2, 1)), np.ones((N/2, 1))])],
> > +                   axis=1)
> > +X = X.T
> > +y = y[:, 0]
> > +A = np.array([[1, 0, 0], [0, 1, 0]])
> > +
> > +def test_NCA():
> > +	# Training
> > +	nca = NCA.NCA(metric=A, method='BFGS', objective='KL-divergence', options={'maxiter': 10, 'disp': True})
> > +	print nca.score(X, y)
>
> sklearn aims Py3 compliance, so you should use function-like print.
>
> Also, how can this test fail other that NCA will throw an exception? You
> may want to add some assertions.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scikit-learn/scikit-learn/pull/5276/files#r39604544>.
>